plugins/src/main/java/com/google/fhir/gateway/plugin/PatientAccessChecker.java

```diff
@@ -336,9 +367,26 @@ public AccessChecker create(
         DecodedJWT jwt,
         HttpFhirClient httpFhirClient,
         FhirContext fhirContext,
-        PatientFinder patientFinder) {
+        PatientFinder patientFinder,
+        BundleEntryPatientFinder bundleEntryPatientFinder) {
       return new PatientAccessChecker(
-          fhirContext, getPatientId(jwt), patientFinder, getSmartFhirPermissionChecker(jwt));
+          fhirContext,
+          getPatientId(jwt),
+          patientFinder,
+          getSmartFhirPermissionChecker(jwt),
+          bundleEntryPatientFinder);
+    }
+  }
+
+  private class BundleAccessDecision {
+    private boolean allBundleOperationsAllowed = true;
+
+    void setBundleEntryDecision(boolean isAccessGranted) {
+      allBundleOperationsAllowed = allBundleOperationsAllowed && isAccessGranted;
```

Comment:
Is there a reason not to stop early? I mean once a bundle entry is not allowed, why do we want to continue?

Reply:
Modified the Bundle processing to incorporate early exit.